### PR TITLE
Create test for checking Publish branch in git extension functionality

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
According to our test plan we should check publish branch in git extension and implement scenario like:

1. Create a workspace.
2. Clone a git project and create a branch with UI.
3. Publish the branch and check this on remote repo.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17289

### CODE REVIEW

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by dmytro-ndp** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[crw-ci-test]

File: tests/e2e/tests/e2e/GitPublishBranch.spec.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+const gitPlugin: GitPlugin = e2eContainer.get(CLASSES.GitPlugin);
+const testWorkspaceUtils: TestWorkspaceUtil = e2eContainer.get<TestWorkspaceUtil>(TYPES.WorkspaceUtil);
+
+
+suite('Publish branch in git extension', async () => {
+    const workspacePrefixUrl: string = `${TestConstants.TS_SELENIUM_BASE_URL}/dashboard/#/ide/${TestConstants.TS_SELENIUM_USERNAME}/`;
+    const wsNameGitPublishBranch = WorkspaceNameHandler.generateWorkspaceName('checkGitPublishBranch-', 5);
+    const changedFile = 'README.md';
+    const branchName = WorkspaceNameHandler.generateWorkspaceName('checkGitPublishBranch', 5);
+    const file = 'https://github.com/' + TestConstants.TS_GITHUB_TEST_REPO + '/blob/' + branchName + '/README.md';
```
Comment (Ohrimenko1988) [inline tests/e2e/tests/e2e/GitPublishBranch.spec.ts]: It is more convenient to use:
```
const file = `https://github.com/${TestConstants.TS_GITHUB_TEST_REPO}/blob/${branchName}/README.md`;
```

File: tests/e2e/tests/e2e/GitPublishBranch.spec.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        await gitPlugin.waitChangedFileInChagesList(changedFile);
+        await gitPlugin.typeCommitMessage(this.test!.title + currentDate);
+        await gitPlugin.commitFromCommandMenu();
+        await gitPlugin.pushChangesFromCommandMenu();
+        await driverHelper.waitAndClick(By.xpath(`//button[@class='theia-button main']`));
+        await gitPlugin.waitDataIsSynchronized();
+        await testWorkspaceUtils.cleanUpAllWorkspaces();
+
+        await driverHelper.navigateToUrl(file);
+        const rawDataFromFile: string = await driverHelper.waitAndGetText(By.xpath(readmeFileContentXpath));
```
Comment (Ohrimenko1988) [inline tests/e2e/tests/e2e/GitPublishBranch.spec.ts]: Better to use another approach,
use locator with the expecting text inside and just wait for visibility this locator by using
```
driverHelper.waitVisibility
```

besides maybe better to create method in proper pageobject?

**Comment by SkorikSergey** [comment]
[crw-ci-test]

**Review by Ohrimenko1988 — APPROVED** [review]

**Pull request merged**